### ISSUE TITLE
Sensor data wrongly used?

### ISSUE THREAD

**Issue by VincenzoKoestler** [Olen/homeassistant-plant#154]
Hello, so far the component is working fine, however I have registered odd behavior.

1) Component is showing data as "unavailable" although the corresponding sensor is properly polling data
2) Data of the sensor used as source for the plant component shows different information that the plant component which should use the sensor data 1:1 no?
<img width="1071" alt="Screenshot 2024-04-10 at 00 01 04" src="https://github.com/Olen/homeassistant-plant/assets/19927591/bc833ac2-dfc6-4c6c-82b0-1db35e678d54">
Below you can see the sensor data polling information and above you can see the plant component not importing the data and showing completely different amounts.

**Comment by Olen** [comment]
Please check the attributes in developer tools that you are indeed watching the right source.
Try to use the change sensor service to set the correct source

**Comment by VincenzoKoestler** [comment]
> Please check the attributes in developer tools that you are indeed watching the right source. Try to use the change sensor service to set the correct source

indeed, you were right; I have created the plant countless times via the UI so I am 100% confident I chose the right sensor, however, in this case the external sensor was set as "banane" not "banane_wz". When I created the plant for the first time, I might have used "banane" only so maybe it kept using that value? I dont have another explanation. But manually changing it via the developer tools fixed the issue for me. I really appreciate it! Thanks a lot :)
